src/controllers/userController.js

```diff
@@ -81,4 +84,66 @@ export default class UserController {
       return Response.error(res, code.serverError, error);
     }
   }
+
+  /**
+     * @description user signUp method
+     * @param {object} req
+     * @param {object} res
+     * @returns {object} createUser
+     * @memberof userController
+     */
+  static async signup(req, res) {
+    const {
+      firstName, lastName, email,
+    } = req.body;
+    // check if user exists
+    const userExist = await UserService.findUser({ email: req.body.email });
+    if (userExist) {
+      return Response.error(res, code.conflict, 'User already exist');
+    }
+    // create the user
+    try {
+      const { password } = req.body;
+      const newPassword = await Password.encryptPassword(password);
+      // update data
+      req.body.password = newPassword;
+      const token = generateToken(req.body);
+
+      const createUser = await models.User.create(req.body);
```

Comment:
I think it will much better if you use service to create a user instead of using models in the controller.